tuf/ngclient/updater.py

```diff
@@ -0,0 +1,578 @@
+# Copyright 2020, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""TUF client workflow implementation.
+"""
+
+import fnmatch
+import logging
+import os
+from typing import Dict, Optional
+from urllib import parse
+
+from securesystemslib import exceptions as sslib_exceptions
+from securesystemslib import hash as sslib_hash
+from securesystemslib import util as sslib_util
+
+from tuf import exceptions
+from tuf.ngclient._internal import (
+    download,
+    requests_fetcher,
+    trusted_metadata_set,
+)
+from tuf.ngclient.config import UpdaterConfig
+from tuf.ngclient.fetcher import FetcherInterface
+
+logger = logging.getLogger(__name__)
+
+# Classes
+class Updater:
+    """
+    An implemetation of the TUF client workflow.
+    Provides a public API for integration in client applications.
+    """
+
+    def __init__(
+        self,
+        repository_dir: str,
+        metadata_base_url: str,
+        target_base_url: Optional[str] = None,
+        fetcher: Optional[FetcherInterface] = None,
+        config: Optional[UpdaterConfig] = None,
+    ):
+        """
+        Args:
+            repository_dir: Local metadata directory. Directory must be
+                writable and it must contain at least a root.json file.
+            metadata_base_url: Base URL for all remote metadata downloads
+            target_base_url: Optional; Default base URL for all remote target
+                downloads. Can be individually set in download_target()
+            fetcher: Optional; FetcherInterface implementation used to download
+                both metadata and targets. Default is RequestsFetcher
+
+        Raises:
+            OSError: Local root.json cannot be read
+            RepositoryError: Local root.json is invalid
+        """
+        self._dir = repository_dir
+        self._metadata_base_url = _ensure_trailing_slash(metadata_base_url)
+        if target_base_url is None:
+            self._target_base_url = None
+        else:
+            self._target_base_url = _ensure_trailing_slash(target_base_url)
+
+        # Read trusted local root metadata
+        data = self._load_local_metadata("root")
+        self._trusted_set = trusted_metadata_set.TrustedMetadataSet(data)
+
+        if config is None:
+            self.config = UpdaterConfig()
+        else:
+            self.config = config
```

Comment:
```suggestion
        self.config = config or UpdaterConfig()
```
should work in this case?